File: Programmers/Lv0/배열 만들기3.py
# 내 풀이
def solution(arr, intervals):
    result = []
    for x, y in intervals:
        result = result + arr[x:y + 1]

    return result

# 다른 사람 풀이

def solution(arr, intervals):
    s1, e1 = intervals[0]
    s2, e2 = intervals[1]
    return arr[s1:e1+1] + arr[s2:e2+1]

File: Programmer_Java/Lv0/배열 만들기 3.java
// package Programmers;
// public class 배열 만들기 3 {
  
// }

// 출처:프로그래머스,
// https://school.programmers.co.kr/learn/courses/30/lessons/181895

// 풀이과정 

/*
 * import java.util.*;

class Solution {
    public int[] solution(int[] arr, int[][] intervals) {
        List<Integer> num_list = new ArrayList<>();
        
        for(int i = 0; i< intervals.length; i++ ){
            int start = intervals[i][0];
            int end = intervals[i][1];
            
            for(int k = start; k<=end; k++){
                num_list.add(arr[k]);
            }
        
        }
        // for(int t = 0; t<num_list.size(); t++){
        //     System.out.println(num_list.get(t));
        // }
        
        int[] answer = new int[num_list.size()];
        
        for(int num = 0; num<num_list.size(); num++){
            answer[num] = num_list.get(num);
        }
        return answer;
    }
}
 */